torch/csrc/jit/import.cpp

```diff
@@ -324,37 +316,23 @@ std::pair<std::shared_ptr<script::Module>, std::string> ModuleDecoder::parseFull
   return std::make_pair(module_lookup(vec), std::move(last));
 }
 
-ModuleDecoder::ModuleDecoder(
-    ModuleLookup module_lookup,
-    std::istream& in) :
-    stream_reader_(&in) {
-  auto model_proto = onnx::ModelProto();
-  auto record = stream_reader_.getLastRecord();
-  model_proto.ParsePartialFromArray(std::get<0>(record).get(), std::get<1>(record));
-  auto graph_proto = model_proto.graph();
-
-  std::unordered_map<std::string, at::Tensor*> param_map;
-
-  for (auto &tensor_proto : graph_proto.initializer()) {
-    std::shared_ptr<script::Module> parent_module;
-    std::string name;
-    std::tie(parent_module, name) = parseFullName(module_lookup, tensor_proto.name());
-
-    auto param = buildParameter(tensor_proto);
-    parent_module->register_parameter(name, param, /* is_buffer = */ tensor_proto.int64_data(0));
-    param_map[tensor_proto.name()] = parent_module->parameter_slot(name);
-  }
-
-  for (auto &node_proto : graph_proto.node()) {
-    std::shared_ptr<script::Module> parent_module;
-    std::string name;
-    std::tie(parent_module, name) = parseFullName(module_lookup, node_proto.name());
-
+MethodDecoder::MethodDecoder(
+    const onnx::ModelProto& model_proto,
+    const std::unordered_map<std::string, at::Tensor*>& param_map,
+    script::Module* parent_module,
+    std::unordered_map<uint64_t, std::shared_ptr<at::Storage>>* storage_map,
+    PyTorchStreamReader* stream_reader) {
+  storage_map_ = storage_map;
+  stream_reader_ = stream_reader;
+  const auto& graph_proto = model_proto.graph();
+  for (const auto &node_proto : graph_proto.node()) {
     std::vector<at::Tensor*> member_inputs;
-    for (auto &param_name : node_proto.input()) {
-      member_inputs.push_back(param_map[param_name]);
+    const std::string& name = node_proto.name();
+    for (const auto &param_name : node_proto.input()) {
+      auto it = param_map.find(param_name);
+      JIT_ASSERT(it != param_map.end());
+      member_inputs.push_back(it->second);
     }
-
     auto graph = buildGraph(node_proto.attribute(0).g());
     // has_domain field has a string iff the method was optimized
     parent_module->set_optimized(node_proto.has_domain());
```

Comment:
@zdevito - are we going to store this flag in serialized TorchScript?

Reply:
No plans to. It is a property of the Module not of the methods on the module, so it makes more sense to store it with the module.